### PR TITLE
Upgrade guidance updates from VLT-172

### PR DESCRIPTION
**Pull request by mladlow**
Trying to clarify some upgrade questions. Learn update to follow in separate PR.

### CODE REVIEW

File: website/content/docs/upgrading/index.mdx [hunk 1]
```diff
@@ -20,11 +20,25 @@ structure that make the data incompatible with a downgrade. If you need to roll
...
+
 ## Learn
 
 Refer to the [Vault Upgrade Standard Procedure](https://learn.hashicorp.com/tutorials/vault/sop-upgrade)
 guide for step-by-step examples of upgrading Vault Enterprise.
 
+## Agent
+
+The Vault Agent is an API client of the Vault Server. Vault APIs are almost
+always backwards compatible. When they are not, this is called oout in the
```
Comment (ncabatoff) [inline website/content/docs/upgrading/index.mdx:36]:
```suggestion
always backwards compatible. When they are not, this is called out in the
```

File: website/content/docs/upgrading/index.mdx [hunk 2]
```diff
@@ -20,11 +20,25 @@ structure that make the data incompatible with a downgrade. If you need to roll
...
+upgrade guide for the new Vault version, and there is a lengthy deprecation
+period. The Vault Agent version can lag behind the Vault Server version, though
+we recommend keeping all versions up to date with the most minor Vault version
```
Comment (ncabatoff) [inline website/content/docs/upgrading/index.mdx:39]: This is a little confusing, what does keeping up to date with the most minor Vault version mean?

Comment (mladlow) [reply website/content/docs/upgrading/index.mdx:39]: Fair. First of all I'm missing a word - "most recent minor Vault version". Does that make the sentence feel like it has enough value, or would you be interested in further re-tooling?

Comment (mladlow) [reply website/content/docs/upgrading/index.mdx:39]:
```suggestion
we recommend keeping all Vault instances up to date with the most recent minor Vault version
```

File: website/content/docs/upgrading/index.mdx [hunk 3]
```diff
@@ -115,8 +129,11 @@ Upgrading installations of Vault which participate in [Enterprise Replication](/
 - When satisfied with functionality of upgraded secondary instances, upgrade
   the primary instance
 
-Upgrades of multiple replicated clusters can be complicated. Here is an
-example.
+~> **Note:** It is not safe to replicate from a newer version of Vault to an
+older version. When upgrading replicated clusters, ensure that upstream clusters
+are always on older versions of Vault than downstream clusters.
```
Comment (ncabatoff) [inline website/content/docs/upgrading/index.mdx:134]:
```suggestion
this is a test
```

Comment (mladlow) [reply website/content/docs/upgrading/index.mdx:134]: Going to take "this is a test" at face value.

**Review by yhyakuna — APPROVED** [review]
LGTM

**Pull request merged**